build-system/amp.extern.js

```diff
@@ -648,6 +648,27 @@ let BindEvaluateBindingsResultDef;
  */
 let BindEvaluateExpressionResultDef;
 
+/////////////////////////////
+////// amp-geo externs
+/////////////////////////////
+
+/**
+ * Actual filled values for this exists in
+ * extensions/amp-geo/0.1/amp-geo.js
+ * @enum {number}
+ */
+const GEO_IN_GROUP = {};
+
+/** 
+  * @typedef {{ 
+  *   ISOCountry: string, 
+  *   matchedISOCountryGroups: !Array<string>, 
+  *   allISOCountryGroups: !Array<string>, 
+  *   isInCountryGroup: (function(string):GEO_IN_GROUP), 
+  * }} 
+  */ 
+let GeoDef; 
```

Comment:
do you want these fields preserved or are they OK to be mangled during minification?

Reply:
I created the issue because I saw
https://github.com/ampproject/amphtml/blob/370b6e3a504537b5e5db4b5881dd095d282de2c4/extensions/amp-geo/0.1/amp-geo.js#L207-L208

But since ISOCountryGroups is removed from `GeoDef`. Is it better to change the above line to `config['ISOCountryGroups']` and allow other fields to be mangled instead?

Reply:
why and where is it removed?

Reply:
@erwinmombay Not sure where to go here suggestions?

Reply:
@zhouyx so everything should be manglable at this point. `ISOCountryGroups` is no longer a thing correct? If that is the case, it LGTM

Reply:
I think we should be good as long as GeoDef is used everywhere and nothing actually needs to be publicly externalized to a 3p (postMessage, XHR, iframe)

Reply:
I agree. I couldn't find where GeoDef is directly exposed to 3p. So we should be good to remove it from the extern.

Another thing is we need to change how ampGeo read the configuration. @jpettitt Let me know if you want me to fix that instead. Thanks.

Reply:
I've reverted the externs - what do we need to change on the configuration?

Reply:
I think 
https://github.com/ampproject/amphtml/blob/95fc028e82c3ab96cc5cfde419f4f20cfa5c333d/extensions/amp-geo/0.1/amp-geo.js#L215-L216
and 
https://github.com/ampproject/amphtml/blob/95fc028e82c3ab96cc5cfde419f4f20cfa5c333d/extensions/amp-geo/0.1/amp-geo.js#L339
need to use `config['ISOCountryGroups']` and `config['AmpBind']`, because we need to preserve the fileNames to read inline config. @erwinmombay to confirm.

Reply:
Are you talking about the global AMP config?

This config is the am-geo configuration, it comes from parsing the `application/json` script.  If that is the issue it probably makes sense to rename config here to differentiate.

Reply:
Talked offline. We decided to avoid using dot fieldName to avoid singlepass mangling the value there.